torch/sparse/_semi_structured_conversions.py

```diff
@@ -0,0 +1,344 @@
+import torch
+
+import torch._inductor.config as cfg
+
+@torch.compile
```

Comment:
Not sure we can turn this on by default. I think for now we'll need to rely on the user to apply this at a top level. Unless we now have enabled composition of torch.compile. cc @jcaip

Reply:
What do you mean, that `@torch.compile` should not be applied here?

Reply:
The user can apply torch.compile to their whole model that also uses this sparse format. torch.compile can then optimize the whole thing. 

It's fine to apply this here, if we can stack this. If a user can make use of these conversion routines and then also still apply torch.compile on top of this torch.compile'd conversion function and get the same benefit.

Reply:
FWIW, following compiles to single Triton kernel:
```Python
import torch

@torch.compile
def myadd(x, y):
    return x * y

@torch.compile
def myaddand2(x, y):
    return myadd(x, y) + 2

x = torch.ones((5,), dtype=torch.float, device="cuda")
y = torch.ones((5,), dtype=torch.float, device="cuda")
print(myaddand2(x, y))
```
So it seems at least in some cases, `@torch.compile` is stacking.

Reply:
Please remove this and re-apply in a separate PR, which will be slated for discussion.